### PR TITLE
move initialization of preprocessing to Preprocess, change initializa…

### PR DESCRIPTION
**Pull request by mschwoer**
move initialization of preprocessing to Preprocess, change initialization of the preprocessing_info

### CODE REVIEW

File: alphastats/DataSet.py [hunk 1]
```diff
@@ -249,44 +266,15 @@ def load_metadata(self, file_path: Union[pd.DataFrame, str]) -> pd.DataFrame:
...
-            PreprocessingStateKeys.CONTAMINATIONS_REMOVED: False,
-            PreprocessingStateKeys.CONTAMINATION_COLUMNS: self.filter_columns,
-            PreprocessingStateKeys.NUM_REMOVED_PG_DUE_TO_CONTAMINATION: 0,
-            PreprocessingStateKeys.DATA_COMPLETENESS_CUTOFF: 0,
-            PreprocessingStateKeys.NUM_PG_REMOVED_DUE_TO_DATA_COMPLETENESS_CUTOFF: 0,
-            PreprocessingStateKeys.MISSING_VALUES_REMOVED: False,
-        }
-        return preprocessing_dict
-
-    def overview(self):
```
Comment (JuliaS92) [inline alphastats/DataSet.py:284]: I would not delete this. However, I would rename it to __repr__ to follow conventions and add a todo to make this more useful.

File: alphastats/DataSet.py [hunk 2]
```diff
@@ -126,8 +137,16 @@ def preprocess(
...
+        self._create_matrix()
+        self.preprocessing_info = Preprocess.init_preprocessing_info(
+            num_samples=self.mat.shape[0],
+            num_protein_groups=self.mat.shape[1],
+            intensity_column=self.intensity_column,
+            filter_columns=self.filter_columns,
+        )
+
+        self.preprocessed = False
+        # TODO fix bug: metadata is not reset/reloaded here
```
Comment (JuliaS92) [inline alphastats/DataSet.py:149]: I think this would fit well in this branch. If you already addressed this ona nother one, just ignore the comment.

**Review by JuliaS92 — APPROVED** [review]
LGTM

TODO: Add a separate attribute in which the settings as passed to preprocess can be stored.

File: alphastats/DataSet.py (hunk 2 above)
Comment (mschwoer) [reply alphastats/DataSet.py:149]: this is not straightforward, so I'd leave it to a later PR

File: alphastats/DataSet.py (hunk 1 above)
Comment (mschwoer) [reply alphastats/DataSet.py:284]: hm, any ideas how to make this more useful?

**Pull request merged**